lib/api/apiUtils/object/deleteObject.js

```diff
@@ -1,9 +1,25 @@
 const data = require('../../../data/wrapper');
 
-function dataDelete(objectGetInfo, log, cb) {
-    data.delete(objectGetInfo, log, err => {
+function dataDelete(locations, method, log, cb) {
+    if (!Array.isArray(locations) || locations.length === 0) {
```

Comment:
I think in earlier versions of the product, `location` in metadata could be a string instead of an array, but not sure if we have to support this here. @rahulreddy an idea?

Reply:
It's good to support the old location format as well. I think at this point it's safe but you can never be too sure.

Reply:
Okay, I'll move forward with merging and forward porting this PR while also working toward a follow-up, smaller PR to support older locations here.